libobs-d3d11/d3d11-subsystem.cpp

```diff
@@ -1237,6 +1190,54 @@ static ULONG GetSdrWhiteNits(HMONITOR monitor)
 	return nits;
 }
 
+gs_monitor_color_info gs_device::GetMonitorColorInfo(HMONITOR hMonitor)
+{
+	IDXGIFactory1 *factory1 = factory;
+	if (!factory1->IsCurrent()) {
+		InitFactory();
+		factory1 = factory;
+		monitor_to_hdr.clear();
+	}
+
+	for (const std::pair<HMONITOR, gs_monitor_color_info> &pair :
+	     monitor_to_hdr) {
+		if (pair.first == hMonitor)
+			return pair.second;
+	}
+
+	ComPtr<IDXGIAdapter> adapter;
+	ComPtr<IDXGIOutput> output;
+	ComPtr<IDXGIOutput6> output6;
+	for (UINT adapterIndex = 0;
+	     SUCCEEDED(factory1->EnumAdapters(adapterIndex, &adapter));
+	     ++adapterIndex) {
+		for (UINT outputIndex = 0;
+		     SUCCEEDED(adapter->EnumOutputs(outputIndex, &output));
+		     ++outputIndex) {
+			if (SUCCEEDED(output->QueryInterface(&output6))) {
+				DXGI_OUTPUT_DESC1 desc1;
+				if (SUCCEEDED(output6->GetDesc1(&desc1)) &&
+				    (desc1.Monitor == hMonitor)) {
+					const bool hdr =
+						desc1.ColorSpace ==
+						DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020;
+					return monitor_to_hdr
+						.emplace_back(
+							hMonitor,
+							gs_monitor_color_info(
+								hdr,
+								desc1.BitsPerColor,
+								GetSdrWhiteNits(
+									desc1.Monitor)))
+						.second;
+				}
+			}
+		}
+	}
```

Comment:
Just a minor nitpick, is there any way to unsquish this code at all?

Reply:
Either `continue` statements or helper functions. Which do you prefer?

Reply:
I push a little tidier version without doing either.